apis/r/tests/testthat/test-SOMAExperiment-query-matrix-outgest.R

```diff
@@ -164,7 +164,10 @@ test_that("matrix outgest assertions", {
   # joinds are used as default dimnames if no obs/var_index is provided
   expect_identical(
     dimnames(query$to_sparse_matrix("X", "counts")),
-    list(as.character(query$obs_joinids()), as.character(query$var_joinids()))
+    lapply(
+      list(query$obs_joinids(), query$var_joinids()),
+      function(x) as.character(x$as_vector())
```

Comment:
Why do we need both `$as_vector()` and `as.character()`?

Reply:
DelayedArray (dependency of SingleCellExperiment) defines `as.character.Array()`; this conflicts with casting of `arrow::Array` objects to character. Arrow only defines `as.character.ArrowDatum()` as its the parent class of `arrow::Array`; R's method resolution selects `DelayedArray:::as.character.Array()` over `arrow:::as.character.ArrowDatum()` as "Array" is a better match for "Array" than "ArrowDatum"

Reply:
Nice catch!